cli/cmd/addon_publish.go

```diff
@@ -0,0 +1,245 @@
+package cmd
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+
+	cloudquery_api "github.com/cloudquery/cloudquery-api-go"
+	"github.com/cloudquery/cloudquery/cli/internal/auth"
+	"github.com/spf13/cobra"
+)
+
+const (
+	addonPublishShort = "Publish to CloudQuery Hub."
+	addonPublishLong  = `Publish to CloudQuery Hub.
+
+This publishes an addon version to CloudQuery Hub from a manifest file.
+`
+	addonPublishExample = `
+# Publish an addon version from a manifest file
+cloudquery addon publish /path/to/manifest.json v1.0.0`
+)
+
+func newCmdAddonPublish() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "publish manifest.json v1.0.0 [--finalize]",
+		Short:   addonPublishShort,
+		Long:    addonPublishLong,
+		Example: addonPublishExample,
+		Hidden:  true,
+		Args:    cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Set up a channel to listen for OS signals for graceful shutdown.
+			ctx, cancel := context.WithCancel(cmd.Context())
+
+			sigChan := make(chan os.Signal, 1)
+			signal.Notify(sigChan, syscall.SIGTERM)
+
+			go func() {
+				<-sigChan
+				cancel()
+			}()
+
+			return runAddonPublish(ctx, cmd, args)
+		},
+	}
+	cmd.Flags().BoolP("finalize", "f", false, `Finalize the addon version after publishing. If false, the addon version will be marked as draft=true.`)
+
+	return cmd
+}
+
+type ManifestJSONV1 struct {
+	Type      string `json:"type"` // always "addon"
+	TeamName  string `json:"team_name"`
+	AddonName string `json:"addon_name"`
+	// AddonType   string   `json:"addon_type"`   // unused
+	// AddonFormat string   `json:"addon_format"` // unused
```

Comment:
These fields belong to the `addon` entry and not `addon_version`, so there's no use for them here. We could try to `GET` the addon and compare/verify the values if we really want to make sure.

Reply:
Seems fine, though I would probably still read the properties, but just not use them